db/db.sql

```diff
@@ -28,6 +29,40 @@ CREATE TABLE `job_links` (
   `bbox` varchar(255),
   `createdAt` datetime not null,
   `updatedAt` datetime not null,
-  FOREIGN KEY(jobID) REFERENCES jobs(jobID));
+  FOREIGN KEY(jobID) REFERENCES jobs(jobID)
+);
 
+CREATE TABLE `work_items` (
+  `id` integer not null primary key autoincrement,
+  `jobID` char(36) not null,
+  `workflowStepIndex` integer not null,
+  `scrollID` varchar(32),
+  `serviceID` varchar(255) not null,
+  `status` text check (`status` in ('ready', 'running', 'successful', 'failed', 'canceled')) not null,
+  `stacCatalogLocation` varchar(255),
+  `createdAt` datetime not null,
+  `updatedAt` datetime not null,
+  FOREIGN KEY(jobID) REFERENCES jobs(jobID)
+  FOREIGN KEY(jobID, workflowStepIndex) REFERENCES workflow_steps(jobID, stepIndex)
```

Comment:
This comment made me realize we needed to add a unique constraint on jobID and stepIndex, so I added that.

While a reference to the PK is likely the better way to set it up, in practice I don't think there's a big difference with using the ID from the table vs jobID and stepIndex as long as there is an index (though the index will be slightly larger). One tiny advantage of storing the step number directly rather than an ID reference in the work items table is that you can just run one select to see the order the items were executed in without having to join to the workflow_steps table to look up the step number. Let me now if it bugs you and we can change things.